### PR TITLE
Fix net_pkt_pull() to remove empty net_bufs

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

File: subsys/net/ip/net_pkt.c [hunk 1]
```diff
@@ -1874,13 +1874,21 @@ int net_pkt_pull(struct net_pkt *pkt, size_t length)
 	struct net_pkt_cursor *c_op = &pkt->cursor;
 	struct net_pkt_cursor backup;
 
+	if (!pkt->frags) {
```
Comment (tbursztyka) [inline subsys/net/ip/net_pkt.c:1877]: pkt->buffer

**Review by tbursztyka — APPROVED** [review]
just a tiny comment

Out of curiosity, what's the use case that triggered this?

**Comment by jukkar** [comment]
> Out of curiosity, what's the use case that triggered this?

For TCP2, I am planning to change how we store the data that is waiting ACKs. So the idea is to have data, that is sent, in a one list of net_buf's. Then when we receive ACKs, we just call net_pkt_pull() and the amount of acked data. Then it is quite easy to implement retransmission using this single net_buf list.

**Comment by tbursztyka** [comment]
> Then it is quite easy to implement retransmission using this single net_buf list.
Indeed, a quick memory optimization

**Pull request merged**